sunpycube/cube/NDCube.py

```diff
@@ -0,0 +1,385 @@
+
+        unit_x_axis: `astropy.units.Unit`
+            The unit of x axis for 2D plots.
+
+        unit_y_axis: `astropy.units.Unit`
+            The unit of y axis for 2D plots.
+
+        unit: `astropy.unit.Unit`
+            The data is changed to the unit given or the cube.unit if not given, for 1D plots.
+
+        axis_ranges: list of physical coordinates for array or None
+            If None array indices will be used for all axes.
+            If a list it should contain one element for each axis of the numpy array.
+            For the image axes a [min, max] pair should be specified which will be
+            passed to :func:`matplotlib.pyplot.imshow` as extent.
+            For the slider axes a [min, max] pair can be specified or an array the
+            same length as the axis which will provide all values for that slider.
+            If None is specified for an axis then the array indices will be used
+            for that axis.
+        """
+        axis_data = ['x' for i in range(2)]
+        axis_data[image_axes[0]] = 'x'
+        axis_data[image_axes[1]] = 'y'
+        if self.data.ndim >= 3:
+            plot = _plot_3D_cube(self, image_axes=axis_data, unit_x_axis=unit_x_axis, unit_y_axis=unit_y_axis,
+                                 axis_ranges=axis_ranges, *kwargs)
+        elif self.data.ndim is 2:
+            plot = _plot_2D_cube(self, axes=axes, image_axes=axis_data, **kwargs)
+        elif self.data.ndim is 1:
+            plot = _plot_1D_cube(self, unit=unit, origin=origin)
+        return plot
+
+    def __getitem__(self, item):
+        if item is None or (isinstance(item, tuple) and None in item):
+            raise IndexError("None indices not supported")
+        data = self.data[item]
+        # here missing axis is reversed as the item comes already in the reverse order
+        # of the input
+        wcs, missing_axis = sunpycube.wcs_util._wcs_slicer(
+            self.wcs, copy.deepcopy(self.missing_axis[::-1]), item)
+        if self.mask is not None:
+            mask = self.mask[item]
+        else:
+            mask = None
+        result = NDCube(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                        meta=self.meta, unit=self.unit, copy=False, missing_axis=missing_axis)
+        return result
+
+
+class NDCubeOrdered(NDCube):
+    """
+    Class representing N dimensional cubes with oriented WCS.
+    Extra arguments are passed on to NDData's init.
+
+    Attributes
+    ----------
+    data: `numpy.ndarray`
+        The spectral cube holding the actual data in this object.
+
+    wcs: `sunpycube.wcs.wcs.WCS`
+        The WCS object containing the axes' information. The axes'
+        priorities are time, spectral, celestial. This means that if
+        present, each of these axis will take precedence over the others.
+        For example, in an x, y, t cube the order would be (t,x,y) and in a
+        lambda, t, y cube the order will be (t, lambda, y).
```

Comment:
The discussion of the ordering of the axes should moved to the top of the docstring as it doesn't matter what order the axes are input, they are reordered as part of initialisation.